sdk/eventhub/event-hubs/review/event-hubs.api.md

```diff
@@ -110,6 +109,7 @@ export class EventHubProducerClient {
     getEventHubProperties(options?: GetEventHubPropertiesOptions): Promise<EventHubProperties>;
     getPartitionIds(options?: GetPartitionIdsOptions): Promise<Array<string>>;
     getPartitionProperties(partitionId: string, options?: GetPartitionPropertiesOptions): Promise<PartitionProperties>;
+    send(eventData: EventData[], options?: SendOptions): Promise<void>;
```

Comment:
How about using an overload of sendBatch and leave "send" to some possible cooler stuffs for the future?
Semantically, we send the whole list as a batch, which could be oversized.

Reply:
We have gone through the exercise of finding cooler send() alternatives extensively and I dont see us going back to it. I like the association of the word "batch" with the actual batch object and would prefer not to use it for array of events

Reply:
@chradek, @bterlson, Any concerns with keeping `sendBatch` for the actual dedicated batch class and using `send` for array of events?

Reply:
Post over an hour long discussion with @bterlson, below is the summary

- Argument for sendBatch:
     - Since using an array of events also end up in us creating a batch under the hood and sending the batch, it helps if the method name reflects the same.
- Argument against sendBatch:
      - A glance at the apis on the EventHubProducerClient shows a reader createBatch and sendBatch leading them to believe that they have to create a batch to be able to send any events.
- Complications for sendBatch implementation:
    - when passing batch object, either throw error if partitionKey or partitionId is passed to sendBatch as well or
	  - throw error when partitionKey is passed
	  - throw error when partitionId is passed and batch was created with partitionKey
    - need to have overloads that take an options bag but of different interfaces or live with the fact that we are now showing options that if set, will result in an error

Reply:
Update: 

The discovery argument against sendBatch() can be countered with the fact that once you start typing sendBatch(), you will be greeted with overloads which should aid with discovery. So, we will go with sendBatch() which will throw validation errors in below cases:
- sendBatch(any-batch, {partitionKey})  // unless partitionKey is the same as that used in batch
- batch = await createBatch({partitionId});
   sendBatch(batch, {some-other-partitionId}) 
- batch = await createBatch({partitionKey});
   sendBatch(batch, {partitionId}) 
- sendBatch(anything, {partitionId, partitionKey})

Reply:
Python will have something similar to this.